### PR TITLE
Amazon Push

### PR DESCRIPTION
**Pull request by JohnOKane**
All the Amazon Android Unity changes in one global PR.

### CODE REVIEW

**Comment by JohnOKane** [comment]
Private SDK PR: https://github.com/Swrve/swrve-sdk/pull/1419
Android SDK PR: https://github.com/Swrve/swrve-android-sdk/pull/246
@Sergio-Mira @seaders @dominicmarmionswrve

File: native/android/SwrveSDKPushSupport/src/amazon/java/com/swrve/unity/adm/SwrveAdmIntentService.java [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+
+public class SwrveAdmIntentService extends ADMMessageHandlerBase {
+    private final static String TAG = "SwrveAdm";
+    private final static String AMAZON_RECENT_PUSH_IDS = "recent_push_ids";
+    private final static String AMAZON_PREFERENCES = "swrve_amazon_unity_pref";
+
+    private final int MAX_ID_CACHE_ITEMS = 16;
+
+    //SwrveMessageReceiver listens for messages from ADM
+    public static class SwrveAdmMessageReceiver extends ADMMessageReceiver {
```
Comment (Sergio-Mira) [inline native/android/SwrveSDKPushSupport/src/amazon/java/com/swrve/unity/adm/SwrveAdmIntentService.java]: Could this be moved to its own file?

File: native/android/SwrveSDKPushSupport/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,38 +1,8 @@
...
-
-        <service android:name="com.swrve.unity.gcm.SwrveGcmInstanceIDListenerService"
-            android:exported="false">
-            <intent-filter>
-                <action android:name="com.google.android.gms.iid.InstanceID"/>
-            </intent-filter>
-        </service>
-        <!-- -->
-    </application>
+       <uses-permission android:name="android.permission.WAKE_LOCK" />
```
Comment (Sergio-Mira) [inline native/android/SwrveSDKPushSupport/src/main/AndroidManifest.xml]: Can be removed and left in the specific flavour if needed

**Comment by Sergio-Mira** [comment]
🚀

**Pull request merged**